doc/man7/EVP_PKEY-MLKEM.pod

```diff
@@ -0,0 +1,102 @@
+=pod
+
+=head1 NAME
+
+EVP_PKEY-ML-KEM-512,
+EVP_PKEY-ML-KEM-768,
+EVP_PKEY-ML-KEM-1024,
+EVP_KEYMGMT-ML-KEM-512,
+EVP_KEYMGMT-ML-KEM-768,
+EVP_KEYMGMT-ML-KEM-1024,
+EVP_PKEY-MLKEM
+- ML-KEM keytype and algorithm support
+
+=head1 DESCRIPTION
+
+The B<ML-KEM-512>, B<ML-KEM-768>, and B<ML-KEM-1024> keytypes are implemented in
+OpenSSL's default provider
+=for comment (TODO(ML-KEM): Add FIPS support).
+
+=head2 Keygen Parameters
+
+By default, no parameters are required for generating a key pair.
+
+=over 4
+
+=item "seed" (B<OSSL_PKEY_PARAM_MLKEM_SEED>) <octet string>
+
+ML-KEM internally requires the generation of a keypair using a random value (seed).
+This optional parameter can be used to set the value prior to key pair generation.
+According to FIPS 203, section 3.3, this parameter should only be used for
+test purposes and be treated with the same care as private key material.
+The length of the seed is 64 bytes.
+
+See L<provider-keymgmt(7)/Common Information Parameters> for further information.
+
+=back
+
+Use EVP_PKEY_CTX_set_params() after calling EVP_PKEY_keygen_init().
+
+=head2 Common parameters
+
+In addition to the common parameters that all keytypes should support (see
+L<provider-keymgmt(7)/Common parameters>), the implementation of these keytypes
+support the following.
+
+=over 4
+
+=item "pub" (B<OSSL_PKEY_PARAM_PUB_KEY>) <octet string>
+
+The public key value.
+
+=item "priv" (B<OSSL_PKEY_PARAM_PRIV_KEY>) <octet string>
```

Comment:
Settable only once, if the key is currently blank. Once either a public or private key has been loaded, the keys become immutable and no further updates to change the value or type (public vs. private) are allowed.